### ISSUE TITLE
ome-zarr spec issues

### ISSUE THREAD

**Issue by will-moore** [ome/omero-ms-zarr#77]
A couple of outstanding issues with the current spec from https://github.com/ome/omero-ms-zarr/issues/76#issuecomment-722279765 that we may want to fix before "v0.2" since they are breaking changes.

- Plate has `"plate":{}` and a Well has `"well":{}` but an Image has `"omero":{}, "multiscales":{}`, so maybe that `omero` key should be renamed to `image` to be consistent?
- Labels should have an object with 'path', e.g. `"labels": [{"path": "0"}, {"path": "1"}] ` to be consistent with elsewhere, instead of `["0", "1"]`

**Comment by sbesson** [comment]
certainly support the idea of unifying the way hierarchical groups refer to their sub-elements. Using a list of JSON objects where each object contains a `path` key has been used in a few places already:


```json
"<supergroup>": {
  "<subgroup>s": [
    {
      "path": "path/to/subgroup/0"
    },
    {
      "path": "path/to/subgroup/1"
    }
  ]
```
 

```json
"<subgroup>": {
}  
```

My main concern with the proposal above is that the content of the current `omero` specification is very driven by OMERO rendering settings. I am not convinced we want to support these concepts as they exist mid-term in an abstract `image` specification. 

Maybe a first step would be to identify the concepts we could see migrating from one specification to the other. The most straightforward candidate is probably the `name` field. Do you see the `image` specification aware of the `multiscales` specification? Or should these be kept independent?

**Comment by will-moore** [comment]
OK so we could go for this as a first step.
```
"image": {
  "name": "my image",
  "multiscales": [
    {
        "datasets": [
            {"path": "0"},
            {"path": "1"},
            {"path": "2"},
            {"path": "3"}
        ],
        "version": "0.1"
    }
  ]
}
```
At least this means that *adding* to this wouldn't be a breaking change.
I think a lot of the data in the 'omero' metadata is not very OMERO specific. E.g. we are currently using it in napari and vizarr to view ome-zarr data (set rendering settings etc) without OMERO being involved (although the data has come from OMERO).
Currently we have:
```
"omero": {
        "channels": [
            {
                "active": true,
                "coefficient": 1.0,
                "color": "FF0000",
                "family": "linear",
                "inverted": false,
                "label": "Red",
                "window": {
                    "end": 255.0,
                    "max": 255.0,
                    "min": 0.0,
                    "start": 0.0
                }
            }
    ],
    "id": 1,
    "rdefs": {
            "defaultT": 0,
            "defaultZ": 0,
            "model": "color"
    }
```
Most useful from this is 'channel: label, color, active` and `default Z/T`.

**Comment by manics** [comment]
I think the above split of `image` and `omero` makes sense as a first step. `image` contains items that are absolutely required to load the image, whereas `omero` contains presentation information that in principle could be ignored.

**Comment by sbesson** [comment]
Seconding @manics comment, I would think the delineation above makes some sense. And I also think it brings us to the conversation  during the OME 20202 community meeting about trying to express different layers of metadata (access, locating in space, visualization, analysis, reproducibility). 

Isn't the wrapping of the `multiscales` spec within an `image` key constituting a breaking change?

**Comment by manics** [comment]
Maybe getting slightly off topic, but regarding the channel name, it's effectively a name for the C-index. Extending this further you might want to have labels for all `C` and `T` indicies (e.g. the `T` label could be the timestamp or time-offset)

**Comment by will-moore** [comment]
Yes, this is certainly a breaking change. Which is why it's good to do this sooner rather than later.
Same for labels change.

**Comment by joshmoore** [comment]
One json-ld based option which wouldn't break multiscales (though we might need to re-evaluate "label-image" and "plate" would be:

```
{
     "@context": "https://specs.openmicroscopy.org/image/v0.1",
     "@type": "Image",
     "@id": "https://s3.embassy.ebi.ac.uk/idr/zarr/v0.1/6001240.zarr", // or UUID?
     "multiscales": [
       ...
     ],
```

i.e. make use of the special fields to say:

 * `@context`: "You can find definition of types here"
 * `@type`: "I am a ..."
 * `@id`: "This is the identifier that will be used to reference me from other locations in this collection of files.

<details><summary>further fields might look like...</summary>

```
  // (bio)schema.org fields
  "includedInDataset": "https://idr.openmicroscopy.org/#current",
  "mainEntity": {
    "@id": "http://disprot.org/DP00086",
    "@type": "Protein",
    "http://purl.org/dc/terms/conformsTo": "https://bioschemas.org/specifications/Protein/0.9-DRAFT",
    "hasRepresentation": "MEEPQSDPSVEPPLS...",
    "identifier": "DP00086",
    "name": "Cellular tumor antigen p53",
    "sameAs": [
      "https://www.uniprot.org/uniprot/P04637",
      "https://mobidb.org/P04637"
    ],

```

</details>

A downside of this as opposed to, e.g. having a top-level `{ "ome": ... }` element is that it requires a general agreement to follow this pattern.  I don't _think_ anyone will conflict with the `@`-keys but something to be aware of. A plus is that these fields can also be arrays/dictionaries.

**Comment by joshmoore** [comment]
Though looking at `multiscales`, I imagine `rendering` should also be a list `Renderings`. cc: @will-moore @jburel

**Comment by will-moore** [comment]
Discussed today - these changes will not come before https://github.com/ome/omero-ms-zarr/pull/75 is merged and spec is tagged.
